### PR TITLE
OpInfo tests with torchinductor

### LINKED ISSUE #727: Use opinfo testing in TorchInductor
We should start using opinfo testing to get more coverage of ops in inductor.

One example of a case where that might help:

> It's possible to use to use `torch.testing._internal.common_methods_invocations.sample_inputs_roll` to get the arguments for testing.
> Then it would match what's being tested in PyTorch in `test/test_ops.py -k roll` tests.

_Originally posted by @IvanYashchuk in https://github.com/pytorch/torchdynamo/pull/725#discussion_r939668338_

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Fixes #727
This PR adds ~5000 unit tests with op_db ... 
And test the entire stack: dynamo + aotautograd + tracing +  decomposition + inductor...

### CODE REVIEW

**Review by desertfire — COMMENTED** [review]
Nice work, so I guess you are adding more tests to the skip list based on the errors seen on CI?

**Comment by SherlockNoMad** [comment]
> Nice work, so I guess you are adding more tests to the skip list based on the errors seen on CI?

Yes, I will iterator on this, and start unskipping and fixing the failures.

**Comment by voznesenskym** [comment]
> > Nice work, so I guess you are adding more tests to the skip list based on the errors seen on CI?
> 
> Yes, I will iterator on this, and start unskipping and fixing the failures.

If you like, wrap all of them in a try-catch-skip, dump the list at the end of the test run, and then we can distribute the work across the whole team / anyone interested?

**Comment by SherlockNoMad** [comment]
> > > Nice work, so I guess you are adding more tests to the skip list based on the errors seen on CI?
> > 
> > 
> > Yes, I will iterator on this, and start unskipping and fixing the failures.
> 
> If you like, wrap all of them in a try-catch-skip, dump the list at the end of the test run, and then we can distribute the work across the whole team / anyone interested?

This is already implemented following the ExpectedTest pattern. Devs can set  PYTORCH_COLLECT_EXPECT=1 to collect the failing tests. 

Yea, we should distribute the work and track it somewhere...

File: .github/workflows/test-inductor.yml [hunk 1]
```diff
@@ -18,6 +18,7 @@ jobs:
     - run: make setup_nightly
     - run: make develop
     - run: pytest ./test/test_torchinductor.py
+    - run: pytest ./test/test_torchinductor_opinfo.py
```
Comment (desertfire) [inline .github/workflows/test-inductor.yml:21]: You need to add a CircleCI task as well, https://github.com/pytorch/torchdynamo/blob/main/.circleci/config.yml . We are not running GPU instance for GHA yet.

Comment (SherlockNoMad) [reply .github/workflows/test-inductor.yml:21]: This is already ran in CircleCI, see https://app.circleci.com/pipelines/github/pytorch/torchdynamo/2764/workflows/67bcf48e-3722-4216-ab6d-fb487b7e97b3/jobs/19887

In CircleCI, it runs pytest on root . All tests will be automatically included.

Comment (desertfire) [reply .github/workflows/test-inductor.yml:21]: NVM, you are right.

**Review by desertfire — APPROVED** [review]

File: test/test_torchinductor_opinfo.py [hunk 2]
```diff
@@ -0,0 +1,707 @@
...
+            # not exercised in test_ops_gradients atm.  The problem is not
+            # complex32 per-se (which is supported by data movement only ops)
+            # but that when we do backwards we expect other ops like add to work
+            and not dtype == torch.complex32
+        )
+        samples = op.sample_inputs(device, dtype, requires_grad=requires_grad)
+
+        for sample_input in samples:
+            args = [sample_input.input] + list(sample_input.args)
+            # kwargs = sample_input.kwargs
```
Comment (ngimel) [inline test/test_torchinductor_opinfo.py:676]: how come we don't need kwargs?

Comment (SherlockNoMad) [reply test/test_torchinductor_opinfo.py:676]: We will need to use this... 
Currently, test_torchinductor's check_model() and check_model_cuda() are not supporting kwargs... Will fix this in the coming PRs.

**Pull request merged**